### PR TITLE
Add support for AWS_S3_FORCE_PATH_STYLE and AWS_ENDPOINT env variables

### PR DESCRIPTION
**Pull request by bithavoc**
adds support for S3-compatible uploads #15

### CODE REVIEW

**Comment by fdr** [comment]
Are there any tests you think we ought to add?

**Comment by bithavoc** [comment]
@fdr can you give me an idea how could we test this? the Configure method doesn't return the inferred aws config where I could be checking the correct values have been assigned based on the environment variables, unless we have a private method build the aws config and we check it against environment variables.

**Comment by fdr** [comment]
Might be nice to be able to connect environment variables to a properly set up struct somehow.

File: upload.go [hunk 1]
```diff
@@ -82,6 +82,13 @@ func Configure() (*TarUploader, *Prefix, error) {
 		return nil, nil, errors.Wrapf(err, "Configure: failed to get AWS credentials; please specify AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY")
 	}
 
+	if endpoint := os.Getenv("AWS_ENDPOINT"); endpoint != "" {
+		config.Endpoint = aws.String(endpoint)
+	}
+
+	s3ForcePathStyle, _ := strconv.ParseBool(os.Getenv("AWS_S3_FORCE_PATH_STYLE"))
```
Comment (fdr) [inline upload.go]: I see you ignore the error here...doesn't it need to be reported to the user somehow?

Comment (bithavoc) [inline upload.go]: I don't think so, the default for `config.S3ForcePathStyle` is false, `strconv.ParseBool` parses correctly: `true`, `false`, `1`, `0`. other than `true`  or `1` is false which is a good default.

Comment (fdr) [inline upload.go]: So if the user types "tru" we should interpret it as "false"? Seems not so great.

Comment (fdr) [inline upload.go]: or the not entirely uncommon "on" vs. "off".

**Comment by x4m** [comment]
This worked for me and is very use full for testing. Thanks.

**Comment by tbartelmess** [comment]
This would be great

**Comment by fdr** [comment]
Let's make the alteration to the error handling so typos or other renditions of "true" do not result in a "false" and commit it.

**Comment by x4m** [comment]
I'll fix bool parsing in following commit.

**Pull request merged**